File: go/205. Isomorphic Strings.go
func isIsomorphic(s string, t string) bool {
	mapping := make(map[byte]byte)
	used := make(map[byte]bool)
	for i := 0; i < len(s); i++ {
		if _, ok := mapping[s[i]]; !ok {
			if used[t[i]] {
				return false
			}
			mapping[s[i]] = t[i]
			used[t[i]] = true
		} else {
			if mapping[s[i]] != t[i] {
				return false
			}
		}
	}
	return true
}

File: java/205. Isomorphic Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        return helper(s,t) && helper(t,s);
    }
    private boolean helper(String s, String t) {
        char[] mapping = new char[255];
        if (s.length() != t.length()) return false;
        for (int i = 0; i < s.length(); i++) {
            int c = s.charAt(i);
            if (mapping[c] == 0) mapping[c] = t.charAt(i);
            else if (mapping[c] != t.charAt(i)) return false;
        }
        return true;
    }
}
